### ISSUE TITLE
[GCP] Default GPU image does not support the latest PyTorch

### ISSUE THREAD

**Issue by DavidePaglieri** [skypilot-org/skypilot#2692]
Since yesterday I have been receiving this error here on a training script that I have used many times before without problems in the past weeks.

```
Could not load library libcudnn_cnn_train.so.8. Error: /usr/local/cuda/lib64/libcudnn_cnn_train.so.8: undefined symbol: _ZN5cudnn3cnn5infer22queryClusterPropertiesERPhS3_, version libcudnn_cnn_infer.so.8

[...]

File "/home/gcpuser/optim/main.py", line 36, in train
loss.backward()
File "/opt/conda/lib/python3.10/site-packages/torch/_tensor.py", line 492, in backward
torch.autograd.backward(
File "/opt/conda/lib/python3.10/site-packages/torch/autograd/__init__.py", line 251, in backward
Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
```

Is it possible that some configuration in the GCP image that skypilot uses has changed? 

This is the configuration of sky_config.yaml I'm using:
```
name: optimizer

resources:
    accelerators: T4:1
    cloud: gcp
    use_spot: true
    spot_recovery: FAILOVER

num_nodes: 1

file_mounts:
    /checkpoint:
        name: $MY_BUCKET_NAME
        mode: MOUNT

    ~/optim:
        name: $MY_BUCKET_NAME
        source: ./optim
        persistent: false
        mode: COPY

setup: |
    set -e  # Fail if any command failed.

    # Fill in your wandb key: copy from https://wandb.ai/authorize
    echo export WANDB_API_KEY=$KEY>> ~/.bashrc

    mkdir -p /checkpoint/
    mkdir -p /checkpoint/wandb

    cd ~/optim
    pip3 install --upgrade pip
    pip3 install -r requirements.txt

run: |
    cd ~/optim

    optimizer="Adam"

    python main.py \
        --resume \
        --num_epochs 40 \
        --wandb_project cifar10_resnet18 \
        --optimizer $optimizer \
        --model_dir /checkpoint/ \
        --wandb_dir /checkpoint/ \
        --model_filename "$optimizer".pth \
        --run_id $optimizer
```

**Comment by concretevitamin** [comment]
Thanks for the report @DavidePaglieri.

From https://github.com/skypilot-org/skypilot-catalog/tree/master/catalogs/v5/gcp it seems that the GCP images.csv hasn't been updated since 3 months ago, so not sure if that's the issue.

cc @Michaelvll

**Comment by Michaelvll** [comment]
Hey @DavidePaglieri, would you mind sharing your requirements.txt? It might be possible that you have some unpinned dependencies that have some new version released recently and require later version of CUDA runtime.

**Comment by DavidePaglieri** [comment]
Hi @Michaelvll, here's my requirements.txt

```
torch>=2.0.1
torchaudio>=2.0.2
torchvideo>=0.0.0
torchvision>=0.15.2
wandb>=0.12.4
tqdm>=4.62.3
```

**Comment by Michaelvll** [comment]
It seems torch's latest version `2.1.0` (released 7 days agao) uses the cuda runtime 12.1 as default, while the previous `2.0.1` uses 11.8. As mentioned in the pytorch doc, you can specify a link to the `11.8` version with `pip install -r requirements.txt --find-links https://download.pytorch.org/whl/cu118`. 

We will look into upgrading the GCP image's cuda version to match the latest requirement

**Comment by DavidePaglieri** [comment]
I confirm that using torch==2.0.1 I no longer have that problem.

**Comment by Michaelvll** [comment]
Reopening this issue as another user encoutered this issue again. We should update the GCP image to support the latest cuda driver and pytorch.

**Comment by concretevitamin** [comment]
Another use ran into this when trying out https://github.com/skypilot-org/skypilot/blob/master/llm/vllm/serve-openai-api.yaml

**Comment by Michaelvll** [comment]
This should be fixed by #2694 and https://github.com/skypilot-org/skypilot-catalog/pull/49. Closing this issue now.
